profiles/permissions.py

```diff
@@ -19,4 +20,9 @@ def has_object_permission(self, request, view, profile):
         if request.method in SAFE_METHODS:
             return True
 
+        if profile.account_privacy == Profile.PRIVATE:
+            # disbaled permission denied for profile with account_privacy = private,
+            # so that we can raise 404 exception.
+            return True
+
```

Comment:
I do not think this is the right place either: this check should go in a specific new permission.